include/wobbly/geometry.h

```diff
@@ -0,0 +1,352 @@
+                    array (points.data ()),
+                    offset (index * 2)
+                {
+                }
+
+                PointView (NumericType *points,
+                           std::size_t index) :
+                    array (points),
+                    offset (index * 2)
+                {
+                }
+
+                PointView (PointView <NumericType> &&view) :
+                    array (view.array),
+                    offset (view.offset)
+                {
+                }
+
+                PointView &
+                operator= (PointView <NumericType> &&view) noexcept
+                {
+                    if (this == &view)
+                        return *this;
+
+                    array = view.array;
+                    offset = view.offset;
+
+                    return *this;
+                }
+
+                /* Copy constructor and assignment operator enabled only if the view
+                 * is truly just observing and has no write access to the point,
+                 * see detail::PointViewCopyabilityBase */
+                PointView (PointView <NumericType> const &p) :
+                    array (p.array),
+                    offset (p.offset)
+                {
+                }
+
+                friend void swap (PointView <NumericType> &first,
+                                  PointView <NumericType> &second)
+                {
+                    using std::swap;
+                    swap (first.array, second.array);
+                    swap (first.offset, second.offset);
+                }
+
+                PointView &
+                operator= (PointView <NumericType> const &p)
+                {
+                    PointView <NumericType> copy (p);
+                    swap (*this, copy);
+                    return *this;
+                }
+
+                /* cppcheck thinks this is a constructor, so we need to suppress
+                 * a constructor warning here.
+                 */
+                // cppcheck-suppress uninitMemberVar
+                operator PointView <NTC> ()
+                {
+                    /* The stored offset is an array offset, but the constructor
+                     * requires a point offset. Divide by two */
+                    return PointView <NTC> (array, offset / 2);
+                }
+
+                template <size_t Offset>
+                NumericType const & get () const
+                {
+                    static_assert (Offset < 2, "Offset must be < 2");
+                    return array[offset + Offset];
+                }
+
+                template <size_t Offset>
+                typename std::enable_if <!(IC::value) && (Offset + 1), void>::type
+                set (NumericType value)
+                {
+                    static_assert (Offset < 2, "Offset must be < 2");
+                    array[offset + Offset] = value;
+                }
+
+            private:
+
```

Comment:
Nitpick: The newline again here.

Reply:
The newline is part of the style of the file, I think.